docs/Workshop.md

```diff
@@ -0,0 +1,774 @@
+  zookeeper-3:
+    image: confluentinc/cp-zookeeper:latest
+    environment:
+      ZOOKEEPER_SERVER_ID: ???
+      ZOOKEEPER_CLIENT_PORT: ???
+      ZOOKEEPER_TICK_TIME: 2000
+      ZOOKEEPER_INIT_LIMIT: 5
+      ZOOKEEPER_SYNC_LIMIT: 2
+      ZOOKEEPER_SERVERS: ???
+```
+
+##### Exercise 1
+
+This Docker Compose configuration sets up a ZooKeeper cluster with three nodes: `zookeeper-1`, `zookeeper-2`, and `zookeeper-3`. Let's explain the key elements:
+
+`version: "3.5"`: Specifies the Docker Compose file version.
+
+`services`: Defines the list of services to be created.
+
+`zookeeper-1, zookeeper-2, zookeeper-3`: Each service represents an individual ZooKeeper node. The numbers at the end of the service names (`-1`, `-2`, `-3`) distinguish between the different nodes.
+
+`image: confluentinc/cp-zookeeper:latest`: Specifies the Docker image to be used for the ZooKeeper service. In this case, it uses the latest version of the `confluentinc/cp-zookeeper` image provided by Confluent.
+
+`environment`: Sets environment variables for the ZooKeeper service.
+
+`ZOOKEEPER_SERVER_ID`: Specifies the unique ID for the ZooKeeper node. Each node in the cluster must have a unique ID.
+
+`ZOOKEEPER_CLIENT_PORT`: Defines the port number on which ZooKeeper listens for client connections.
+
+`ZOOKEEPER_TICK_TIME`: Sets the length of a single tick, which is the basic time unit used by ZooKeeper.
+
+`ZOOKEEPER_INIT_LIMIT`: Defines the time (in ticks) that ZooKeeper servers can take to connect and synchronize with each other.
+
+`ZOOKEEPER_SYNC_LIMIT`: Specifies the maximum time (in ticks) that ZooKeeper servers can be out of sync with each other.
+
+`ZOOKEEPER_SERVERS`: Sets the list of ZooKeeper servers in the format `server:id1:host1:port1;server:id2:host2:port2;....` This configuration helps ZooKeeper nodes discover and connect to each other in the cluster.
+
+You need to fill in the ZOOKEEPER_SERVER_ID, ZOOKEEPER_CLIENT_PORT, and ZOOKEEPER_SERVERS environment variables for the zookeeper-3 service to complete the configuration. After filling it in, you can check that the zookeeper cluster is setup appropriately by saving the file, and running:
+
+```bash
+docker-compose up -d
+docker-compose logs -f zookeeper-3
+```
+
+in a terminal window in your clone's `src/` directory that contains the docker-compose.yml.
+
+These commands launch the ZooKeeper containers in the background (-d flag) according to the configuration specified in the `docker-compose.yml` file, and display the logs of the zookeeper node you filled out in the exercise. In the logs you should see no errors and eventually see an info log like following (it may take a few moments):
+
+```shell
+INFO [QuorumPeer[...]]: INFO ... - Accepted socket connection from /<IP_Address>:<Port>
+```
+Congratulations, you've setup your zookeeper cluster. Stop following the logs with (<Cmd|Ctrl> C. You can now tear it down with:
+
+```bash
+docker-compose down
+git reset --hard setup-kafka
```

Comment:
This moves to `setup-kafka`, but the instructions below say, "Copy the following code". So, will `setup-kafka` have the same code as `setup-zookeeper`?